src/transaction/log_page_buffer.c

```diff
@@ -3307,6 +3310,20 @@ logpb_flush_all_append_pages (THREAD_ENTRY * thread_p)
   gettimeofday (&start_time, NULL);
 #endif /* CUBRID_DEBUG */
 
+  if (is_tran_server_with_remote_storage ())
```

Comment:
This is already a huge function and this new bit is an 'early out'. Maybe fork with two new functions? - fork function, remote storage function, local storage function (current function, only renamed).
Just saying...